lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -35,6 +36,22 @@ function dosomething_global_init() {
       drupal_goto($new_url);
     }
   }
+
+  // Handle translation redirects for authenticated and anonymous users when
+  // they request bare (Global English) URLs.
+  //
+  // Passing 'noredirect=1' will defeat the redirect.
+  if (arg(0) == 'node' && is_numeric(arg(1)) && '' == arg(2) &&
+      !dosomething_global_is_translation_request() &&
+      empty($_GET['noredirect'])) {
+
+    if ($user->uid) {
```

Comment:
Bit of Drupal-ese: `$user->uid` is always set -- anonymous users have `$user->uid = 0`.